tests/recovery/blobmanipulator/blobmanipulator.go

```diff
@@ -0,0 +1,288 @@
+		fiofilewriter.MinNumFilesPerWriteField: strconv.Itoa(numFiles),
+	}
+
+	ctx := context.Background()
+
+	_, err = bm.fileWriter.WriteRandomFiles(ctx, fileWriteOpts)
+	if err != nil {
+		return err
+	}
+
+	// create snapshot of the data
+	snapPath := bm.fileWriter.DataDirectory(ctx)
+	log.Printf("Creating snapshot of directory %s", snapPath)
+
+	_, err = bm.TakeSnapshot(snapPath)
+	if err != nil {
+		return err
+	}
+
+	numFiles = 50
+
+	fileWriteOpts = map[string]string{
+		fiofilewriter.MaxDirDepthField:         strconv.Itoa(1),
+		fiofilewriter.MaxFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MinFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MaxNumFilesPerWriteField: strconv.Itoa(numFiles),
+		fiofilewriter.MinNumFilesPerWriteField: strconv.Itoa(numFiles),
+	}
+
+	_, err = bm.fileWriter.WriteRandomFiles(ctx, fileWriteOpts)
+	if err != nil {
+		return err
+	}
+
+	fileSize = 40 * 1024 * 1024
+	numFiles = 1
+
+	fileWriteOpts = map[string]string{
+		fiofilewriter.MaxDirDepthField:         strconv.Itoa(1),
+		fiofilewriter.MaxFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MinFileSizeField:         strconv.Itoa(fileSize),
+		fiofilewriter.MaxNumFilesPerWriteField: strconv.Itoa(numFiles),
+		fiofilewriter.MinNumFilesPerWriteField: strconv.Itoa(numFiles),
+	}
+
+	_, err = bm.fileWriter.WriteRandomFiles(ctx, fileWriteOpts)
+	if err != nil {
+		return err
+	}
+
+	// create snapshot of the data
+	log.Printf("Creating snapshot of directory %s", snapPath)
+
+	_, err = bm.TakeSnapshot(snapPath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TakeSnapshot creates snapshot of the provided directory.
+func (bm *BlobManipulator) TakeSnapshot(dir string) (string, error) {
+	err := bm.KopiaCommandRunner.ConnectRepo("filesystem", "--path="+bm.DataRepoPath)
+	if err != nil {
+		return "", err
+	}
+
+	stdout, _, err := bm.KopiaCommandRunner.Run("snapshot", "create", dir)
+	if err != nil {
+		return stdout, err
+	}
+
+	return "", nil
+}
+
+// SnapshotFixRemoveFilesByBlobID runs snapshot fix remove-files command with a provided blob id.
+func (bm *BlobManipulator) SnapshotFixRemoveFilesByBlobID(blobID string) (string, error) {
+	// Get hold of object ID that can be used in the snapshot fix command
+	output, msg, err := bm.KopiaCommandRunner.Run("snapshot", "fix", "remove-files", "--object-id="+blobID, "--commit")
+	if err != nil {
+		log.Println(output, msg)
+		return output, err
+	}
+
+	return "", nil
+}
+
+// SnapshotFixRemoveFilesByFilename runs snapshot fix remove-files command with a provided file name.
+func (bm *BlobManipulator) SnapshotFixRemoveFilesByFilename(filename string) (string, error) {
+	// Get hold of the filename that can be used to in the snapshot fix command
+	stdout, msg, err := bm.KopiaCommandRunner.Run("snapshot", "fix", "remove-files", "--filename="+filename, "--commit")
+	if err != nil {
+		log.Println(stdout, msg)
```

Comment:
`Run` method I think already logs internally, if I'm not mistaken? Do we need to log again here? Same question for situations above.